lib/pcr.c

```diff
@@ -138,3 +138,13 @@ int pcr_parse_list(const char *str, int len, TPMS_PCR_SELECTION *pcrSel) {
 
     return 0;
 }
+
+int get_max_supported_pcrs(TSS2_SYS_CONTEXT *sapi_context) {
+    TPMI_YES_NO moreData;
+    TPMS_CAPABILITY_DATA capabilityData;
+    UINT32 rval = Tss2_Sys_GetCapability( sapi_context, 0, TPM_CAP_TPM_PROPERTIES, TPM_PT_PCR_COUNT, 1, &moreData, &capabilityData, 0 );
+    if (rval != TPM_RC_SUCCESS) {
+        return -1;
+    }
+    return(capabilityData.data.tpmProperties.tpmProperty[0].value) ; 
```

Comment:
value is a UINT32, so on 32 bit systems, really big values will be negative.

```
/* Table 93  Definition of TPMS_TAGGED_PROPERTY Structure <OUT> */
typedef	struct {
	TPM_PT	property;	 /* a property identifier  */
	UINT32	value;	 /* the value of the property  */
} TPMS_TAGGED_PROPERTY;
```

Please pass in a UINT32 pointer as an output parameter.